pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -498,6 +495,97 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
+func updateObjectstLabels(crLables map[string]string, objs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	var err error
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		// Label objects with version of the operator they were created by
+		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
+
+		labels, err = updateLabels(obj, labels, crLables)
```

Comment:
it seems that `updateLabels/updateAppLabelsFromCR` is not only updating the `labels`, but also changing the `obj.spec.template...` via `updateTemplateLabels`. that seems kind of off to me, since the function is not doing exactly what it is declaring to do.

Reply:
So to update to function name to `updateObjLabels` and return obj instead labels ?

Reply:
updated to updateObjLabels